core/arch/aarch64/emit_utils.c

```diff
@@ -153,6 +216,10 @@ insert_exit_stub_other_flags(dcontext_t *dcontext, fragment_t *f, linkstub_t *l,
         ASSERT(sizeof(app_pc) == 8);
         pc += DIRECT_EXIT_STUB_DATA_SZ / sizeof(uint);
         /* We start off with the fcache-return routine address in the slot. */
+        /* AArch64 uses shared fragments, so fcache_return gencode should be
+         * shared too.
```

Comment:
This is inaccurate: the presence of shared fragments does not mean there are no private fragments.  For ease of mangling, selfmod fragments are thread-private; temp fragments during trace building are thread-private; there used to be some cases of syscall-ending fragments that were thread-private.

The key fact is that A64's private fragments use shared gencode (and in fact A64 never generates private gencode).

Reply:
Thanks. Corrected the comment.